examples/mp-guarantees/src/main.rs

```diff
@@ -365,6 +411,143 @@ fn test_overhead(config: &TestConfig, guarantee: &str, faulty: bool) -> TestResu
+            sys.send_local_message("sender-node", "sender", message);
+        }
+    });
+    if let Err(e) = res {
+        e.print_trace();
+        Err(e.message())
+    } else {
+        Ok(true)
+    }
+}
+
+fn test_mc_message_drops(config: &TestConfig) -> TestResult {
+    let mut sys = build_system(config, false);
+    sys.network().set_drop_rate(0.1);
+    let messages: Vec<Message> = generate_message_texts(&mut sys, 2)
+        .into_iter()
+        .map(|text| Message::new("MESSAGE", &format!(r#"{{"text": "{}"}}"#, text)))
+        .collect();
+    let strategy_config = StrategyConfig::default()
+        .prune(prunes::state_depth(7))
+        .goal(goals::any_goal(vec![
+            goals::got_n_local_messages("receiver-node", "receiver", 2),
+            goals::no_events(),
+        ]))
+        .invariant(mc_invariant_guarantees(messages.clone(), *config));
+    let mut mc = ModelChecker::new::<Bfs>(&sys, strategy_config);
+    let res = mc.run_with_change(move |sys| {
+        for message in messages {
+            sys.send_local_message("sender-node", "sender", message);
+        }
+    });
+    if let Err(e) = res {
+        e.print_trace();
+        Err(e.message())
+    } else {
+        Ok(true)
+    }
+}
+
+fn test_mc_unstable_network(config: &TestConfig) -> TestResult {
+    let mut sys = build_system(config, false);
+    sys.network().set_drop_rate(0.1);
+    sys.network().set_dupl_rate(0.1);
+    let msg_count = if config.ordered { 3 } else { 2 };
+    let messages: Vec<Message> = generate_message_texts(&mut sys, msg_count)
+        .into_iter()
+        .map(|text| Message::new("MESSAGE", &format!(r#"{{"text": "{}"}}"#, text)))
+        .collect();
+    let num_drops_allowed = 1;
+    let num_duplication_allowed = 1;
+    let goal = if config.reliable && config.once {
+        goals::all_goals(vec![
+            goals::got_n_local_messages("receiver-node", "receiver", msg_count),
+            goals::no_events(),
+        ])
+    } else {
+        goals::no_events()
+    };
+    let invariant = if config.ordered {
+        invariants::all_invariants(vec![
+            invariants::state_depth(20),
+            mc_invariant_guarantees(messages.clone(), *config),
+            invariants::time_limit(Duration::from_secs(10)),
+        ])
+    } else {
+        invariants::all_invariants(vec![
+            invariants::state_depth(20),
+            mc_invariant_guarantees(messages.clone(), *config),
+        ])
+    };
+    let strategy_config = StrategyConfig::default()
+        .prune(prunes::any_prune(vec![
+            prunes::events_limit(LogEntry::is_mc_message_dropped, num_drops_allowed),
+            prunes::events_limit(LogEntry::is_mc_message_duplicated, num_duplication_allowed),
+            prunes::events_limit(LogEntry::is_mc_timer_fired, 1),
+            prunes::events_limit(LogEntry::is_mc_message_received, msg_count + num_drops_allowed),
+        ]))
+        .goal(goal)
+        .invariant(invariant);
```

Comment:
Немного доработал и повысил лимит до 60 секунд.

```suggestion
    let mut invariants = vec![
        invariants::state_depth(20),
        mc_invariant_guarantees(messages.clone(), *config),
    ];
    if config.ordered {
        invariants.push(invariants::time_limit(Duration::from_secs(60)))
    };
    let strategy_config = StrategyConfig::default()
        .prune(prunes::any_prune(vec![
            prunes::events_limit(LogEntry::is_mc_message_dropped, num_drops_allowed),
            prunes::events_limit(LogEntry::is_mc_message_duplicated, num_duplication_allowed),
            prunes::events_limit(LogEntry::is_mc_timer_fired, 1),
            prunes::events_limit(LogEntry::is_mc_message_received, msg_count + num_drops_allowed),
        ]))
        .goal(goal)
        .invariant(invariants::all_invariants(invariants));
```

Reply:
ой, да, 10 у меня от тестирования просочилось  :sweat_smile:

Reply:
поднял до 80